src/backend/distributed/README.md

```diff
@@ -1744,9 +1744,141 @@ The reason for handling dependencies and deparsing in post-process step is that
+
+If Citus encounters a dependency it can't support, it will throw an error instead of silently allowing it. The rationale behind this approach is to avoid subtle issues later, especially when adding new nodes. For instance, Citus currently throws an error for circular dependencies. The main function performing these checks is `EnsureDependenciesCanBeDistributed()`.
+
+During the DFS, Citus might need to extend its search, especially for complex dependencies like array types that don't have a straightforward dependency on their element types. Citus expands the traversal to account for such cases. The main function responsible for this is `ExpandCitusSupportedTypes()`, which has extensive comments explaining the specific rules.
+
+### `current_user` vs `superuser` for Object Propagation:
+
+The difference between `MarkObjectDistributed()` and `MarkObjectDistributedViaSuperuser()` is important here. Generally, Citus tries to minimize the use of `superuser` operations for security reasons. However, there are cases where it's necessary. We employ `superuser` permissions primarily when marking the dependencies of an object we are working on. This is because creating dependencies might require higher-level privileges that the `current_user` might not have. For example, if a schema depends on a role, and the `current_user` doesn't have the privilege to create roles, an error will be thrown. To avoid this, we use `superuser` for creating dependencies.
+
+However, there's an exception. If the dependency is created within the same transaction, we use the `current_user`. This prevents visibility issues and is mainly relevant for `serial` columns. More details can be found in [Citus GitHub PR 7028](https://github.com/citusdata/citus/pull/7028).
+
+### When Are the Objects in `pg_dist_object` Used?
+
+There are three main scenarios:
+
++ When adding a new node—or more precisely, activating it—Citus reads all objects listed in `pg_dist_object`, sorts them by dependency, and then creates those objects on the new node. The core C function for this is `SendDependencyCreationCommands()`, and the sorting is done by `OrderObjectAddressListInDependencyOrder()`.
+
++ When Citus creates a new object and processes its dependencies, any dependencies already marked as distributed are skipped. This is handled in `FollowNewSupportedDependencies()`, where dependencies are bypassed if `IsAnyObjectDistributed()` returns true.
+
++ When user modifies an object, Citus acts only when the object is distributed. For non-distributed object, Citus gives the control back to Postgres.
 
```

Comment:
I have been wondering: Is there a world in which we no longer have `pg_dist_object`, but simply propagate everything?

Reply:
I defer this question to @thanodnl 😹